build-system/tasks/visual-diff.js

```diff
@@ -422,51 +417,68 @@ async function snapshotWebpages(percy, page, webpages) {
+    // more network requests. This method is flaky since Puppeteer doesn't
+    // always understand Chrome's network activity, so ignore timeouts again.
+    const pagePromise = page.goto(
+        `${BASE_URL}/${url}`, {waitUntil: 'networkidle0'})
+        .then(() => {}, () => {})
+        .then(async() => {
+          log('verbose', 'Navigation to page',
+              colors.yellow(`${BASE_URL}/${url}`), 'is done, verifying page');
+
+          await page.bringToFront();
+
+          await verifyCssElements(page, url, webpage.forbidden_css,
+              webpage.loading_incomplete_css, webpage.loading_complete_css);
+
+          if (webpage.loading_complete_delay_ms) {
+            log('verbose', 'Waiting',
+                colors.cyan(`${webpage.loading_complete_delay_ms}ms`),
+                'for loading to complete');
+            await sleep(webpage.loading_complete_delay_ms);
+          }
 
-    if (webpage.enable_percy_javascript) {
-      snapshotOptions.enableJavaScript = true;
-      // Remove all scripts that have an external source, leaving only those
-      // scripts that are inlined in the page inside a <script> tag.
-      await page.evaluate(
-          'document.head.querySelectorAll("script[src]").forEach(' +
-          'node => node./*OK*/remove())');
-    }
+          const snapshotOptions = Object.assign({}, DEFAULT_SNAPSHOT_OPTIONS);
 
-    if (viewport) {
-      log('verbose', 'Wrapping viewport-constrained page in an iframe');
-      await page.evaluate(WRAP_IN_IFRAME_SCRIPT
-          .replace(/__WIDTH__/g, viewport.width)
-          .replace(/__HEIGHT__/g, viewport.height));
-      await page.setViewport({
-        width: VIEWPORT_WIDTH,
-        height: VIEWPORT_HEIGHT,
-      });
-    }
+          if (webpage.enable_percy_javascript) {
+            snapshotOptions.enableJavaScript = true;
+            // Remove all scripts that have an external source, leaving only
+            // those scripts that are inlined in the page inside a <script> tag.
+            await page.evaluate(
+                'document.head.querySelectorAll("script[src]").forEach(' +
+                'node => node./*OK*/remove())');
+          }
 
-    await percy.snapshot(name, page, snapshotOptions);
-    await clearExperiments(page);
-    log('travis', colors.cyan('●'));
+          if (viewport) {
+            snapshotOptions.widths = [viewport.width];
+            log('verbose', 'Wrapping viewport-constrained page in an iframe');
+            await page.evaluate(WRAP_IN_IFRAME_SCRIPT
+                .replace(/__WIDTH__/g, viewport.width)
+                .replace(/__HEIGHT__/g, viewport.height));
+            await page.setViewport({
+              width: VIEWPORT_WIDTH,
+              height: VIEWPORT_HEIGHT,
+            });
+          }
+
+          await percy.snapshot(name, page, snapshotOptions);
+          await page.close();
+          log('travis', colors.cyan('●'));
```

Comment:
nit: I've noticed that when an error is encountered, it gets printed inline with these dots. Easy fix: insert a newline before any errors printed during tests.